application/src/main/scala/com/gu/deskcomexporttool/S3InteractionsWriter.scala

```diff
@@ -53,22 +82,68 @@ object S3InteractionsWriter {
             }
           }
 
+          private def createStatusMapping(status: Option[String]): Option[Int] = {
+            status match {
+              case Some("received") => Some(1)
+              case Some("pending") => Some(3)
+              case Some("sent") => Some(3)
+              case _ => None
+            }
+          }
+
+          private def createDirectionMapping(direction: Option[String]) : String = {
+            direction match {
+              case Some("in") => "TRUE"
+              case Some("out") => "FALSE"
+              case _ => ""
+            }
+          }
+
+          private def parseDate(date: String): Either[S3Error, Instant] = {
+            Either.catchNonFatal {
+              Instant.from(DateTimeFormatter.ISO_DATE_TIME.parse(date))
+            }.leftMap(error => S3Error(s"Failed to parse date: $date error:$error"))
+          }
+
+          private def sanitizeUpdatedDate(createdDate: Instant, updatedDate: Instant) = {
+            if(updatedDate.isBefore(createdDate)) {
```

Comment:
I'm assuming this is because the desk.com data can be in an odd state where something is updated before it is created? 😖

Reply:
Correct :)